packages/webcomponents/src/components/payment-transactions-list/payment-transactions-list.tsx

```diff
@@ -0,0 +1,175 @@
+import { makeGetPaymentTransactions } from '../../actions/payment/get-payment-transactions';
+import { PaymentService } from '../../api/services/payment.service';
+import { Table } from '../../utils/table';
+import { TableClickActions } from '../../ui-components/table/event-types';
+import { table, tableCell } from '../../styles/parts';
+import { StyledHost, TableEmptyState, TableErrorState, TableLoadingState } from '../../ui-components';
+import { defaultColumnsKeys, paymentTransactionTableCells, paymentTransactionTableColumns } from './payment-transactions-table';
+
+@Component({
+  tag: 'justifi-payment-transactions-list',
+  shadow: true
+})
+export class PaymentTransactionsList {
+  @State() balanceTransactions: PaymentBalanceTransaction[] = [];
+  @State() transactionsTable: Table<PaymentBalanceTransaction>;
+  @State() isLoading: boolean = true;
+  @State() errorMessage: string = null;
+  @State() paging: PagingInfo = pagingDefaults;
+  @State() pagingParams: any = {};
+
+  @Prop() paymentId: string;
+  @Prop() authToken: string;
+  @Prop() apiOrigin?: string = PROXY_API_ORIGIN;
+  @Prop() columns?: string = defaultColumnsKeys;
+
+  @Event({ eventName: 'click-event', bubbles: true }) clickEvent: EventEmitter<ComponentClickEvent>;
+  @Event({ eventName: 'error-event' }) errorEvent: EventEmitter<ComponentErrorEvent>;
+  
+  analytics: JustifiAnalytics;
+
+  componentWillLoad() {
+    this.transactionsTable = new Table<PaymentBalanceTransaction>(this.balanceTransactions, this.columns, paymentTransactionTableColumns, paymentTransactionTableCells);
+    checkPkgVersion();
+    this.analytics = new JustifiAnalytics(this);
+    this.initializeApi();
+  }
+
+  disconnectedCallback() {
+    this.analytics.cleanup();
+  }
+
+  @Watch('paymentId')
+  @Watch('authToken')
+  propChanged() {
+    this.initializeApi();
+  }
+
+  private handleError(code: ComponentErrorCodes, errorMessage: string, severity: ComponentErrorSeverity) {
+    this.errorEvent.emit({ errorCode: code, message: errorMessage, severity });
+  }
+
+  private initializeApi() {
+    if (this.paymentId && this.authToken) {
+      const getPaymentTransactions = makeGetPaymentTransactions({
+        id: this.paymentId,
+        authToken: this.authToken,
+        service: new PaymentService(),
+        apiOrigin: this.apiOrigin
+      });
+
+      getPaymentTransactions({
+        params: this.pagingParams,
+        onSuccess: ({ balanceTransactions, pagingInfo }) => {
+          this.balanceTransactions = balanceTransactions;
+          this.paging = pagingInfo;
+          this.transactionsTable.collectionData = this.balanceTransactions;
+        },
+        onError: ({ error, code, severity }) => {
+          this.errorMessage = error.message;
+          this.handleError(error, code, severity);
+        },
+        final: () => {
+          this.isLoading = false;
+        }
+      });
+    } else {
+      this.errorMessage = 'Payment ID and Auth Token are required';
```

Comment:
Not sure how our other error messages are written but we should consider using the prop names just to be more clear